Marlin/SdBaseFile.cpp

```diff
@@ -674,7 +674,7 @@ bool SdBaseFile::open(SdBaseFile* dirFile,
       index = 0;
     }
     // initialize as empty file
-    memset(p, 0, sizeof(dir_t));
+    ZERO(p);
```

Comment:
Aha, yes, I see that p is declared as `dir_t *p`. So, you're right! It should be the size of what p points to, and not the `p` pointer itself that's cleared. I forgot this kind of case when crafting that macro. I should have used a `constexpr` function instead of a macro, but I'm only just learning about this new feature of C++.